File: db/event_link.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_link.
*/

class Event_Link extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_link";
    $this->order = "";
    $this->domain = "event";
    $this->field['event_link_id']['type'] = 'SERIAL';
    $this->field['event_link_id']['not_null'] = true;
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['not_null'] = true;
    $this->field['link_type_id']['type'] = 'INTEGER';
    $this->field['link_type_id']['not_null'] = true;
    $this->field['url']['type'] = 'VARCHAR';
    $this->field['url']['length'] = 1024;
    $this->field['url']['not_null'] = true;
    $this->field['title']['type'] = 'VARCHAR';
    $this->field['title']['length'] = 128;
    $this->field['description']['type'] = 'VARCHAR';
    $this->field['description']['length'] = 128;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['event_link_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_link.rb
module Momomoto
  class Event_link < Base
    def initialize
      super
      @domain = 'event'
      @log_changes = true
      @fields = {
        :event_link_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :event_id => Datatype::Integer.new( {:not_null=>true} ),
        :url => Datatype::Varchar.new( {:length=>1024, :not_null=>true} ),
        :rank => Datatype::Integer.new( {} ),
        :title => Datatype::Varchar.new( {:length=>256} ),
        :last_modified => Datatype::Timestamp.new( {:with_timezone=>true, :not_null=>true, :default=>true, :auto_update=>true} ),
        :last_modified_by => Datatype::Integer.new( {} )
      }
    end
  end
end
